Core/Src/main.c

```diff
@@ -87,14 +90,34 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
+  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_SET); //CS default high
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1); //PWM on Channel 1
+
+  uint8_t txData[3] = {0x01, 0x80, 0x00};
+  uint8_t rxData[3] = {0};
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //CS low start communication
+	  HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 3, 1000);
```

Comment:
Could you explain to me why you are choosing to run this function?

Reply:
I chose the HAL_SPI_TransmitReceive one from the .h file because the instructions said that I needed a function that can both send at receive data.

Reply:
Ohh I don't even know that. It did say it in the bootcamp instruction